rust/sealevel/programs/interchain-accounts/src/instruction.rs

```diff
@@ -0,0 +1,155 @@
+//! HelloWorld instructions.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use hyperlane_core::H256;
+use hyperlane_sealevel_connection_client::router::RemoteRouterConfig;
+use hyperlane_sealevel_igp::accounts::InterchainGasPaymasterType;
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::program_storage_pda_seeds;
+
+/// Init instruction data.
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+pub struct Init {
+    /// The local domain.
+    pub local_domain: u32,
+    /// The mailbox.
+    pub mailbox: Pubkey,
+    /// The ISM.
+    pub ism: Option<Pubkey>,
+    /// The IGP.
+    pub igp: Option<(Pubkey, InterchainGasPaymasterType)>,
+    /// The owner.
+    pub owner: Option<Pubkey>,
+}
+
+/// CallRemote instruction data.
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+pub struct CallRemoteMessage {
+    /// The address of the owner on the origin domain.
+    pub sender: Pubkey,
```

Comment:
Any reason to not have this be implied by the accounts passed in? this is the typical pattern on Solana and would save us 32 bytes in the tx as well